src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -7,89 +7,357 @@
+                value = GetInt16();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
+        public bool TryGetInt32(out int value)
+        {
+            try
+            {
+                value = GetInt32();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
+        public bool TryGetInt64(out long value)
+        {
+            try
+            {
+                value = GetInt64();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
+        public bool TryGetSingle(out float value)
+        {
+            try
+            {
+                value = GetSingle();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
+        public bool TryGetDouble(out double value)
+        {
+            try
+            {
+                value = GetDouble();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
         [CLSCompliant(false)]
-        public bool TryGetSByte(out sbyte value) { throw null; }
+        public bool TryGetSByte(out sbyte value)
+        {
+            try
+            {
+                value = GetSByte();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
         [CLSCompliant(false)]
-        public bool TryGetUInt16(out ushort value) { throw null; }
+        public bool TryGetUInt16(out ushort value)
+        {
+            try
+            {
+                value = GetUInt16();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
         [CLSCompliant(false)]
-        public bool TryGetUInt32(out uint value) { throw null; }
+        public bool TryGetUInt32(out uint value)
+        {
+            try
+            {
+                value = GetUInt32();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
         [CLSCompliant(false)]
-        public bool TryGetUInt64(out ulong value) { throw null; }
+        public bool TryGetUInt64(out ulong value)
+        {
+            try
+            {
+                value = GetUInt64();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
+
+        public void SetString(string value)
```

Comment:
This probably needs a better name, like `SetFormattedValue`.

Reply:
Ok, I'll change it.